Cpp math nan (#1084)
Codecademy/docs: content/cpp/concepts/math-functions/terms/rint/rint.md

---
Title: 'rint()'

> Description: 'The rint() function in C++ rounds the argument to an integral value using the current rounding mode.'

Reviewer: ```suggestion
Description: 'Rounds the argument to an integral value using the current rounding mode.'
```

Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
  - 'Methods'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

> 

Reviewer: ```suggestion

The **`rint()`** function in C++ rounds the argument to an integral value using the current rounding mode.

```

## Syntax

```pseudo
rint(x)
```

Argument must be a `double`/`float`/`long double`, and the return value will be same type.

> ## Return value

Reviewer: Remove the "Return Value" header and include the following sentence at the end of the "Syntax" section.

Reviewer: @chiragcode11, This comment has not been resolved.

> The rint() function rounds the argument x to an integral value, using the rounding direction specified by fegetround() and returns the value. By default, the rounding direction is set to 'to-nearest'. The rounding direction can be set to other values using fesetround() function.

Reviewer: ```suggestion
The `rint()` function rounds the argument x to an integral value, using the rounding direction specified by `fegetround()` and returns the value. By default, the rounding direction is set to 'to-nearest'. The rounding direction can be set to other values using `fesetround()` function.
```

## Example

> The following example uses the `rint()` function to find the value of x after rounding off:

Reviewer: ```suggestion
The following example uses the `rint()` function to find the value of `x` after rounding:
```

```cpp
#include <iostream>
#include <cmath>

int main() {
  double x = 11.87;
  double result;
  result = rint(x);
  std::cout << "Rounding to-nearest integer of (" << x << ") = " << result << "/n";
}
```

This produces the following output:

```shell
Rounding to-nearest integer (11.87) = 12
```

## Codebyte Example

> The following example is runnable and returns the nearest value to x

Reviewer: ```suggestion
The following example is runnable and returns the nearest value to `x`:
```
